src/aot/aot.cpp

```diff
@@ -253,16 +253,40 @@ int load(BPFtrace &bpftrace, const std::string &in)
     return 1;
   }
 
-  uint8_t *ptr = static_cast<uint8_t *>(
+  // Start of the ELF file
+  auto *ptr = static_cast<uint8_t *>(
       ::mmap(NULL, in_file_size, PROT_READ, MAP_PRIVATE, infd, 0));
   if (ptr == MAP_FAILED) {
     auto saved_err = errno;
     LOG(ERROR) << "Failed to mmap: " << in << ": " << std::strerror(saved_err);
     return 1;
   }
 
-  // Validate header
-  auto hdr = reinterpret_cast<const Header *>(ptr);
+  // Find .btaot section
+  auto *elf_header = reinterpret_cast<Elf64_Ehdr *>(ptr);
+  auto string_table_header = reinterpret_cast<Elf64_Shdr *>(
+      ptr + elf_header->e_shoff +
+      elf_header->e_shstrndx * elf_header->e_shentsize);
+  auto section_header = reinterpret_cast<Elf64_Shdr *>(ptr +
+                                                       elf_header->e_shoff);
```

Comment:
I think there should probably be some error checking here. Long term, it'd be nice to not have a dependency against libelf. So that the final AOT binaries are lighter. But for now, probably best to use libelf APIs.

Here's a lazy link you can refer to for example:  https://lore.kernel.org/bpf/82ae2ddd83a8d85ae071e650cf80005c36bb1343.1711389163.git.dxu@dxuuu.xyz/   (see btf_encoder__tag_kfuncs())